### PR TITLE
build: Only deploy docs for latest tag

### PR DESCRIPTION
**Pull request by adigitoleo**
This brings us one step towards having docs for every tagged version. For now (after the next tag push) there should only be one website for the latest tagged version. Earlier versions can be built offline. Progress to #199.

In any case, it is better than the current situation: html documentation is being deployed for every single push on `main`, and the docs website did not indicate a version number anywhere. With these GH workflow adjustments, the docs should still be _built_ for each PR/push to main, allowing to catch errors in the documentation setup as they arise, but the html pages will not deployed to GitHub Pages if there is no tag. The version (obtained from `git describe`) is also indicated clearly on the website landing page and in the API docs sidebar footer.

Related things that I rolled in here:
- new make target to build docs (simply `make html` in the action spec)
- favicon and proper branding (multiple resolutions of logo)
- removed a duplicated sentence from the README
- `docs/template/README.md` to explain how to maintain the template
- fixed `rm -r` -> `rm -rf` in `make clean` recipe

### CODE REVIEW

**Comment by adigitoleo** [comment]
> docs should still be built for each PR/push to main

Obviously not happening. Either need to split `docs.yml` into `docs-build.yml` and `docs-deploy.yml`, or somehow get the conditionals to work properly in `docs.yml`.

**Comment by adigitoleo** [comment]
I just realised that based on [this log](https://github.com/seismic-anisotropy/PyDRex/actions/runs/10206968651/job/28240878575?pr=207) maybe I don't need to duplicate the common job steps after all (b2f7d77). I guess the if-check in L63 of docs.yml only evaluates after the "Build" task has completed, so even if we aren't on top of a tag it should still check that `make html` works before aborting. I can remove that commit after review.

File: README.md [hunk 1]
```diff
@@ -45,34 +45,27 @@ declared in the `pyproject.toml` file.
...
-The documentation can be built offline using [pdoc](https://github.com/mitmproxy/pdoc),
-with the command:
+The documentation can be built offline from a git clone or source distribution.
+Install documentation builder dependencies with `pip install '.[doc]'`.
 
-    pdoc -t docs/template -o html --math pydrex tests
-
-which will output the html documentation into a folder called `html`.
+Run the make command `make html` which will output the documentation,
+including the generated API reference, into a folder called `html`.
```
Comment (Patol75) [inline README.md:63]: Run `make html` from the terminal to generate PyDRex's documentation (available in the `html` directory), including the API reference.

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
   push:
     branches:
       - main
+    tags:
+      - v*
   pull_request:
     branches:
       - main
-  workflow_dispatch:
+  workflow_dispatch:  # Allows to manually trigger the job from the web UI.
```
Comment (Patol75) [inline .github/workflows/docs.yml:12]: You should be able to remove the colon after `workflow_dispatch`.

**Review by Patol75 — CHANGES_REQUESTED** [review]
Looking good; some minor comments.

File: .github/workflows/docs.yml (hunk 2 above)
Comment (adigitoleo) [reply .github/workflows/docs.yml:12]: pre-commit fails though, seems like invalid yaml without it:

```
check yaml...............................................................Failed
- hook id: check-yaml
- exit code: 1

while scanning a simple key
  in ".github/workflows/docs.yml", line 12, column 3
could not find expected ':'
  in ".github/workflows/docs.yml", line 14, column 1
```

**Review by Patol75 — APPROVED** [review]

**Pull request merged**